operations/last_revision.go

```diff
@@ -91,13 +106,39 @@ func LastRevision() cli.Command {
 				return errors.New("no matching version found")
 			}
 
-			grip.Infof("Latest version that matches criteria: %s\nRevision: %s\n", utility.FromStringPtr(matchingVersion.Id), utility.FromStringPtr(matchingVersion.Revision))
+			if err := printLastRevision(matchingVersion, jsonOutput); err != nil {
+				return errors.Wrap(err, "printing last revision")
+			}
 
 			return nil
 		},
 	}
 }
 
+func printLastRevision(v *model.APIVersion, jsonOutput bool) error {
+	versionID := utility.FromStringPtr(v.Id)
+	revision := utility.FromStringPtr(v.Revision)
+
+	if jsonOutput {
+		output, err := json.MarshalIndent(struct {
+			VersionID string `json:"version_id"`
+			Revision  string `json:"revision"`
+		}{
+			VersionID: versionID,
+			Revision:  revision,
+		}, "", "\t")
+		if err != nil {
+			return errors.Wrap(err, "marshalling output to JSON")
```

Comment:
would it make sense to also return a json error when the flag is on incase they're using some tool or script that expectes a json?

Reply:
I was thinking that it would make it complicated since the CLI doesn't have a native way to return errors in JSON. If they want to detect errors with a script, they could split stdout (which only contains the JSON result) and stderr (which contains any fatal CLI errors) and check both streams.